src/screens/resetPassword/hooks/index.ts

```diff
@@ -0,0 +1,148 @@
+import { useLocation, useNavigate } from 'react-router-dom';
+import { useState, useEffect, useMemo } from 'react';
+import jwt from 'jwt-decode';
+import { TResetPasswordToken } from './types';
+import useResetPasswordRequest from '../../../service/requests/useResetPasswordRequest';
+import {
+  validatePassword,
+  validateConfirmPassword,
+} from '../../../utils/validateFields';
+import { useSnackBar } from '../../../utils/renderSnackBar';
+import { SCREENS } from '../../../utils/screens';
+
+const useQuery = () => {
+  const { search } = useLocation();
+  return useMemo(() => new URLSearchParams(search), [search]);
+};
+
+const useResetPassword = () => {
+  const query = useQuery();
+  const queryToken = query.get('token');
+
+  const { showErrorSnackBar } = useSnackBar();
+
+  const [token, setToken] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+  const [passwordError, setPasswordError] = useState('');
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+
+  const decodedToken = useMemo(() => {
+    if (token) {
+      return jwt(token) as TResetPasswordToken;
+    }
+
+    return {} as TResetPasswordToken;
+  }, [token]);
+
+  const validToken = useMemo(() => {
```

Comment:
Sempre começa variáveis booleanas com `is` ou `has` para indicar que são booleanas:

```suggestion
  const isValidToken = useMemo(() => {
```